vochain/offchaindatahandler/offchaindatahandler.go

```diff
@@ -140,6 +140,22 @@ func (c *OffChainDataHandler) OnProcessesStart(pids [][]byte) {
 	}
 }
 
+// OnSetAccount is triggered when a new account is created or modifyied. If metadata info is present, it is enqueued.
+func (c *OffChainDataHandler) OnSetAccount(addr []byte, account *vochain.Account) {
+	c.queueLock.Lock()
+	defer c.queueLock.Unlock()
+	if !c.importOnlyNew || !c.isFastSync {
```

Comment:
What if the tx is `AddDelegateForAccount` ? Will this be executed with the same metadata and will enqueue the same InfoURI ?

Reply:
If the file is already downloaded it will do nothing. However you are right, it would be more convenient to check the kind of transaction before adding the URI to the queue. But the event handler does not provide this information, so it should be added.

Reply:
A new issue should be created to fix this, first on the Event handler layer.

Reply:
Task created